### PR TITLE
[Persistent] Cover CustomerData by Unit Test

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

**Increase Test Coverage**

Cover class: `Magento\Persistent\CustomerData\Persistent`

### Manual testing scenarios (*)

vendor/bin/phpunit app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-6469](https://jira.corp.magento.com/browse/ENGCOM-6469) has been created to process this Pull Request

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    /**
+     * Test getSectionData() when customer login and enable persistent
+     */
+    public function testGetSectionDataCustomerLoginAndEnablePersistent()
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(true);
+
+        $persistentSessionMock = $this->createPartialMock(PersistentSession::class, ['getCustomerId']);
+        $persistentSessionMock->method('getCustomerId')->willReturn(1);
```
Comment (lbajsarowicz) [inline app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: You should not use magic numbers - that should be private const with a meaningful name - like `STUB_CUSTOMER_ID`

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        $this->persistentSessionHelperMock->method('getSession')->willReturn($persistentSessionMock);
+
+        $customerMock = $this->createMock(CustomerInterface::class);
+        $this->customerRepositoryMock->method('getById')->with(1)->willReturn($customerMock);
```
Comment (lbajsarowicz) [inline app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: Here you should refer to previously mentioned private const.

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        $this->customerViewHelperMock->method('getCustomerName')->with($customerMock)->willReturn('Adam John');
```
Comment (lbajsarowicz) [inline app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: `Adam John` is another example of magic literal. You should replace that with STUB_CUSTOMER_NAME

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+     */
+    private $persistentSessionHelperMock;
+
+    /**
+     * @var View|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $customerViewHelperMock;
+
+    /**
+     * @var CustomerRepositoryInterface|\PHPUnit_Framework_MockObject_MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: Use alias `\PHPUnit_Framework_MockObject_MockObject` as `MockObject`

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    {
+        $this->persistentSessionHelperMock->method('isPersistent')->willReturn(false);
+
+        $this->assertEquals([], $this->customerData->getSectionData());
+    }
+
+    /**
+     * Test getSectionData() when customer doesn't login
+     */
+    public function testGetSectionDataWithNotLogin()
```
Comment (lbajsarowicz) [inline app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: In my personal opinion - the name of test should be more explicit - Optionally use pattern `given / when / then`. `testGetSectionDataWhenCustomerNotLoggedInReturnsEmptyArray`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]

**Comment by edenduong** [comment]
Hi @lbajsarowicz: Thank you. I have changed the source code. Please check it.

File: app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php (hunk 2 above)
Comment (lbajsarowicz) [reply app/code/Magento/Persistent/Test/Unit/CustomerData/PersistentTest.php]: You've left this unchanged. Replace `1` with `self::STUB_CUSTOMER_ID`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Great improvement. One last thing and I'd love to approve such PR.

**Comment by edenduong** [comment]
Hi @lbajsarowicz: Thank you. It should be ok now :)

**Review by lbajsarowicz — APPROVED** [review]
Thank you for your contribution :+1:

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6469](https://jira.corp.magento.com/browse/ENGCOM-6469) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**